### PR TITLE
[elfio] Update elfio from 3.8 to 3.9

### LINKED ISSUE #17248: [elfio] update to 3.9
Library name: elfio

New version number: 3.9

### PR DESCRIPTION
**Pull request by schultetwin1**
**Describe the pull request**
Updates elfio from 3.8 to 3.9

- #### What does your PR fix?  
  Fixes #17248

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
   Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
The failures caused by popsift will be fixed by #17277.

File: versions/e-/elfio.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "9114641146014eb1c706e939aefa7c72ec8167b8",
+      "version-string": "3.9",
```
Comment (JonLiu1993) [inline versions/e-/elfio.json:5]: please change version-string to version

File: ports/elfio/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "elfio",
-  "version-string": "3.8",
+  "version-string": "3.9",
```
Comment (JonLiu1993) [inline ports/elfio/vcpkg.json:3]: @schultetwin1 ,please change version-string to version

File: versions/e-/elfio.json (hunk 1 above)
Comment (schultetwin1) [reply versions/e-/elfio.json]: Done!

File: ports/elfio/vcpkg.json (hunk 2 above)
Comment (schultetwin1) [reply ports/elfio/vcpkg.json]: Done!

**Comment by strega-nil-ms** [comment]
Thanks @schultetwin1 :)

**Pull request merged**